Browse books by genre

MinaNemr opened the pull request:
#71

@Veegas please review the code
@omarkamal95 please review documentation

Commit:
Removed auto inserted book collection #71

Commit:
Removed from requiring the code to insert books #71

Commit:
added wireframes for the user story #71

Commit:
Merge branch 'Development' of https://github.com/nelsammak/Croma into browse_books_by_Genre

Commit:
added better styling icons #71

Commit:
button contains genre attribute now #71

Commit:
edited text color and position #71

Commit:
added margins #71

Commit:
edited style so text on icons would be smaller #71

Commit:
changed hover image to smaller one #71

Commit:
removed href from html + updated pointer style #71

Commit:
changes styling + removed all icons expect one #71

Commit:
removed hover icon #71

Commit:
removed hover image and underlining from icon #71

Commit:
added new hover icon #71

Commit:
Removed unecessary title #71

Commit:
edited id of html attribute from x to link #71

Commit:
added ng-click directive #71

Commit:
created some dummy data to test genre html #71

Commit:
added ng-repeat directive to test GenreCtrl #71

Commit:
added genre reference in HTML page #71

Commit:
added get function to get JSON file of genres from backend #71

Commit:
added router genre with function to respond with all genres on DB #71

Commit:
required genres model in book view to use it #71

Commit:
added genres model schema and edited route function #71

Commit:
added doucmentation to the function that returns all book genres #71

Commit:
added functionality add genres to genres model automatically after inserting any book

Commit:
added Genre moder inn epub.js #71

Commit:
changed genre attribute to name #71

Commit:
findOneOrCreate plugin to avoid adding duplicates #71

Commit:
replaced adding genre with findoneorcreate #71

Commit:
uninstalled findorcreate #71

Commit:
added remove book genre from db and adding it again to avoid duplicates #71

Commit:
genre.name instead of genre to display anduse gemre name not genre object #71

Commit:
added some margins to books.html #71

Commit:
added documentation #71

Commit:
seperetad genres from book into array of genres and add it to genres model through looping #71

Commit:
fixed split paramaters to match all cases of book genres #71

Commit:
added sequence diagrams #71

omarkamal95 commented on frontend/views/partials/genre.html:
```diff
@@ -4,8 +4,42 @@
 </head>
-<body style="margin-top: 300px">
+<body class="page-top" class="index">
+   <section id="portfolio" >
```
@MinaNemr Just add documentation to the html genre.html, Just for things like where does each section start and end, for example like in index: <!-- start of dashboard --> , then after the closing tag <!-- end of dashboard --> things like this. just a general description of what is in each section and where does it end.

Commit:
added documentation to genre.HTML and removed previous icons #71

MinaNemr commented:
@omarkamal95 I added documentation to genre.html as u said
please re-review and let me know if i need to modify anything.

Veegas commented:
@MinaNemr  Coding conventions passed,
Bas lama ba3mel upload le ketab el server beycrash

Veegas commented:
@MinaNemr  Law el genre kebir shewaya beytla3 bara el box beta3o

MinaNemr commented:
@Veegas  Uploading a book is working on my side with no problems
try to drop ur DB (books & genres) 
i included a code to drop all books and genres in epub.js but it is commented
try again and let me know if my code is the one causing the problems

MinaNemr commented:
@Veegas 
Yeah i knw that already but the only solution is to make the font size smaller and it won't look good
but i don't think that there's a genre name more than 8 letters
please let me know the maximum number of letters i should put into consideration

MinaNemr commented:
@Veegas 
P.S : i changed the way genres is added to a book, this might be the reason causing crash at ur side
check epub.js to see the difference

Veegas commented:
@MinaNemr  The server crashes when an epub with no genre is uploaded, garabha kda w shoof eh el hay7sal w sal7aha

Veegas commented:
We msh lazem testaghar el font momken te3mel 7agat tanya 3ashan akeed fih genre aktar men 8 7oroof

MinaNemr commented on frontend/scripts/controllers/books.js:
```diff
@@ -43,8 +43,26 @@ angular.module('angularPassportApp')
 
 //genre controller's job is  to  take the genre specified from html by click of user and request that genre from backend then route to it 
+/**
+
```
@Veegas

Veegas replied:
@MinaNemr  la2 7ot el documentation fo2 el function 3alatool

Commit:
changned params to param #73

Commit:
fixed doucemntation of GenreCtrl #73

MinaNemr commented on frontend/scripts/controllers/books.js:
```diff
@@ -43,9 +43,32 @@ angular.module('angularPassportApp')
 
 
```
@omarkamal95

MinaNemr commented on backend/views/book.js:
```diff
@@ -305,7 +306,22 @@ module.exports = function(router) {
 
 		})
 	});
```
@omarkamal95

Commit:
fixed documentation (from params to param) #71

MinaNemr commented:
development merged with branch and conflicts resolved

Commit:
fixed some docuementation #71

Commit:
Fixed genre name size bug #71

Now we can insert genre name up to 28 words
14 words per line

Commit:
fixed the bug of inserted a book with no genres

if a book without a genre is inserted into our DB
it would be assigned with unspecified genre

MinaNemr commented:
@Veegas 
long genre name exception is handled by making genre icon hold two lines each can hold up to 14 letters
book without genre bug is fixed by letting the genre (Unspecified) 
Please Review code again

Commit:
seperated genre services into service folder into books.js file #71

Commit:
added documentation to genre services #71

MinaNemr commented:
@omarkamal95 please re-review documentation 
@Mernawaseif please review UMLs
@Israa-Elwan please review scenario

Commit:
change service functions style #71

MinaNemr commented:
please no one review until i fix a bug

Commit:
fixed a bug at inserting any book unspecified genre was inserted at any book even if genre existed #71

Commit:
book remove and genre remove commented #71

MinaNemr commented:
to test code u will need to alter this block in backend/views/epub.js

//code to remove books from DB if we want to
//Book.remove({}, function error (err) {});
//code to remove genres from DB if we want to
//Genre.remove({}, function error (err) {});

TO

//code to remove books from DB if we want to
Book.remove({}, function error (err) {});
//code to remove genres from DB if we want to
Genre.remove({}, function error (err) {});

and run the server once
then alter it again TO

//code to remove books from DB if we want to
//Book.remove({}, function error (err) {});
//code to remove genres from DB if we want to
//Genre.remove({}, function error (err) {});

then add some books then test

Commit:
fixed doucemntation of GenreCtrl #73

Commit:
Merge remote-tracking branch 'origin/Development' into browse_books_by_Genre

Conflicts:
	backend/app.js
	frontend/views/index.html
	frontend/views/partials/books.html

Mernawaseif commented:
@MinaNemr reviewed wireframes and sequence UMLs, no comments on them.

Israa-Elwan commented:
@MinaNemr i reviewed the scenario , don't have any comments on it

omarkamal95 commented:
@MinaNemr Documentation review done.

Veegas commented:
@MinaNemr  Merge el conflicts

Commit:
Merge remote-tracking branch 'origin/Development' into browse_books_by_Genre

Conflicts:
	backend/app.js
	backend/views/book.js
	frontend/index.html
	frontend/scripts/controllers/books.js
	frontend/views/partials/books.html
	frontend/views/partials/genre.html
